### DISCUSSION TITLE
Can't add users to organisation

### DISCUSSION THREAD

**Discussion by BiLLy-85** [dani-garcia/vaultwarden#5319] [Q&A] [answered]
Since a while now i can't add users to organisations anymore.

I have an organisation with 3 users and would like to add another existing user to that organisation but get the following error:

```
Er is een fout opgetreden.
<!DOCTYPE html> <html lang="en"> <head> <meta charset="utf-8"> <meta name="color-scheme" content="light dark"> <title>422 Unprocessable Entity</title> </head> <body align="center"> <div role="main" align="center"> <h1>422: Unprocessable Entity</h1> <p>The request was well-formed but was unable to be followed due to semantic errors.</p> <hr /> </div> <div role="contentinfo" align="center"> <small>Rocket</small> </div> </body> </html>
![image](https://github.com/user-attachments/assets/ac4bf562-ec9d-465b-824d-6948b3d9203c)
```

I'm running vaultwarden on docker on my synology NAS with the reverse proxy of synology.
The log in the docker says amoung others: `[%date and time%][vaultwarden::api::core::organizations::_][WARN] Data guard 'Json < InviteData >' failed: Parse("{\"emails\": [\"%mail@domain.com%\"],\"collections\":null,\"accessAll\":true,\"type\":3,\"permissions\":{\"response\":null}}", Error("missing field 'groups'", line: 1, column: 114)).`

Everything else seem to be working fine, only this.

### Your environment (Generated via diagnostics page)

* Vaultwarden version: v1.32.7
* Web-vault version: v2.25.0
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Database type: SQLite
* Database version: 3.46.0
* Environment settings overridden!: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Websocket Check: true
* HTTP Response Checks: true

### Config & Details (Generated via diagnostics page)

<details><summary>Show Config & Details</summary>

**Environment settings which are overridden:** SIGNUPS_ALLOWED, ADMIN_TOKEN

**Config:**

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_max_note_size": 10000,
  "_smtp_img_src": "***:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_connect_src": "",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://***********",
  "domain_origin": "*****://***********",
  "domain_path": "",
  "domain_set": true,
  "duo_context_purge_schedule": "30 * * * * *",
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "duo_use_iframe": false,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": true,
  "enforce_single_org_with_reset_pw_policy": false,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "http_request_block_non_global_ips": true,
  "http_request_block_regex": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "increase_note_size_limit": false,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Web-kluis",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "*************************",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "push_enabled": true,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "***************",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "*************************",
  "smtp_from_name": "VaultWarden",
  "smtp_host": "**************",
  "smtp_password": "***",
  "smtp_port": 465,
  "smtp_security": "force_tls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "*************************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```

</details>

Please help to resolve this problem.

**Comment by BlackDex** [comment, marked as answer]
Your web-vault is very very old. Make sure you use the latest web-vault.

You have it overridden somewhere.

**Reply by BiLLy-85** [reply]
Oké, interesting...
So, like I've said, i'm using docker on Synology, and there i'm using this image: https://registry.hub.docker.com/r/vaultwarden/server/
Here it say 'Alternative implementation of the Bitwarden server API in Rust, including the Web Vault.'.
Since I've recently updated the image (twice), i assumed everything would have been up to date, but it seems not.

The status page is not saying anything bad about the web-vaults version:
![Screenshot_20241223_093923_Chrome.jpg](https://github.com/user-attachments/assets/90546081-106f-49de-95ac-e11edc94e3ad)

Can you please send me in the right direction on how to update the web-vault?

**Reply by BlackDex** [reply]
Well, it shouldn't have that version of the web-vault. So you must have overridden it somewhere in your container config.

All docker containers have the correct version embedded.

**Reply by BiLLy-85** [reply]
Thank you for that, I did find why it wasn't right;
I had made a volume to /web-vault and one for /web-vault/vw_static.
I do remember why i did that in the past; to translate the e-mails which where send from Vaultwarden.
But now i see these templates are in the data folder, so maybe i did this for the logo's which I wanted to change some day.

After removing those volumes, I still did get an error, only this time something else, somethings like;
[ERROR] Handlebars
template_name: Some("email/send_org_invite.html",),
Line_no: Some(12,),
column_no: Some(65,),
reason: MissingVariable(Some("org_id",),

The user which I tried to add however is now a member of the organisation, it looks now only the invitation mail did not send.

So I've compared my email templates with the current ones from the sourcecode and noticed the URL was made a lot shorter, changed this and the mail was send as well and everything seems to be working again.

Thank you very much for your help!
